docs/tutorial_email_generator.md

```diff
@@ -0,0 +1,225 @@
+        """
+    )
+    response = openai.ChatCompletion.create(model='gpt-3.5-turbo',
+                                            messages=[{
+                                                "role": "system",
+                                                "content": prompt
+                                            }],
+                                            max_tokens=500
+                                            )
+    email_body = response['choices'][0]['message']['content'].strip()
+    return email_body
+
+subject = "Application for the software engineer role"
+recipient_name = "Google"
+content_description = """
+- I want to apply for the role
+- I have many years of experience in AI, making me a strong candidate
+- Ready to start ASAP
+"""
+email = generate_email(subject, recipient_name, content_description)
+print("####")
+print(email)
+print("####")
+
+factual_consistency = langcheck.metrics.factual_consistency(generated_outputs=[email], sources=[content_description])
+print(f"factual consistency = {factual_consistency.metric_values[0]:.2f}")
+```
+
+```text
+####
+Subject: Application for the software engineer role
+
+Dear Hiring Manager,
+
+I am writing to apply for the software engineer role at Google. With many years of experience in AI, I believe I am a strong candidate for this position. I am ready to start as soon as possible.
+
+Thank you for considering my application.
+
+Sincerely,
+[Your Name]
+####
+
+factual consistency = 0.55
+```
+
+## Evaluating with LangCheck's Reference-Based Metrics
+
+The metrics that we have examined so far have been text structure and reference free metrics, meaning that we don’t need a reference output to evaluate the email generator’s outputs. However, if we do have some reference outputs, we would be able to leverage the reference based metrics as well.
+
+For all of the email prompts, I wrote out the emails that I personally would’ve written (see here). Let’s now assume that my email writing skills are good enough to call these the reference outputs, and compute some of LangCheck’s reference based metrics. Below, we compute the `rougeL` and `semantic_similarity` metrics, and show them both in a scatter plot.
+
+```python
+generated_emails = [item["generated_email"] for item in [japan_trip, running_late, google_application]]
+reference_emails = [item["reference"] for item in [japan_trip, running_late, google_application]]
+
+rougeL = langcheck.metrics.rougeL(generated_outputs=generated_emails, reference_outputs=reference_emails)
+semantic_similarity = langcheck.metrics.semantic_similarity(generated_outputs=generated_emails, reference_outputs=reference_emails)
+
+langcheck.plot.scatter(semantic_similarity, rougeL)
+```
+![MetricValue output](_static/email_tutorial_scatter.png)
+
+Due to the longform nature of these generated emails, we can see that the `rougeL` values tend to be a bit low, even though the `semantic_similarity` values are high.
```

Comment:
It might be cool to do something with automatically generating the email subject too.

Reply:
I changed the subject to be auto-generated too by making it follow the format
```
        ###
        Subject: <appropriate subject>

        <Email content>
        ###
```